services/orderService.js

```diff
@@ -0,0 +1,64 @@
+const orderDao = require('../models/orderDao');
+const userDao = require('../models/userDao');
+const cartDao = require('../models/cartDao');
+
+const createOrder = async (userId, address, orderStatusEnum) => {
+  try {
+    const userPoint = await userDao.myAccount(userId);
+    const { carts } = await cartDao.getCarts(userId);
+
+    if (carts.length === 0) {
+      const error = new Error('Cart is empty');
+      error.statusCode = 400;
+      throw error;
+    }
+
+    const cartItems = carts.map((cart) => {
+      const price = cart.price;
+      const weight = cart.weight;
+      const quantity = cart.quantity;
+      const totalPrice = price * quantity;
+      const totalWeight = weight * quantity;
+
+      if (userPoint < totalPrice) {
+        const error = new Error('Not enough points to complete this purchase');
+        error.statusCode = 400;
+        throw error;
+      }
+      return {
+        price,
+        weight,
+        quantity,
+        totalPrice,
+        totalWeight,
+      };
+    });
```

Comment:
1. map 안에서는 값 계산만 해주세요.
2. map 안에서는 scope가 달라집니다. map 밖에서 let으로 totalPrice, totalWeht를 선언해 주세요.
3. 이를 다시 구현 하시면 아래 reduce는 필요하지 않을 겁니다.
```suggestion
    const cartItems = carts.map((cart) => {
      const price = cart.price;
      const weight = cart.weight;
      const quantity = cart.quantity;
      const totalPrice = price * quantity;
      const totalWeight = weight * quantity;

      if (userPoint < totalPrice) {
        const error = new Error('Not enough points to complete this purchase');
        error.statusCode = 400;
        throw error;
      }
      return {
        price,
        weight,
        quantity,
        totalPrice,
        totalWeight,
      };
    });
```